packages/flutter/lib/src/widgets/spacer.dart

```diff
@@ -52,13 +66,49 @@ class Spacer extends StatelessWidget {
   /// children, according to the flex factors of the flexible children.
   ///
   /// Defaults to one.
-  final int flex;
+  final int? flex;
+
+  /// The absolute fixed length in the main axis that will be taken.
+  final double? length;
 
   @override
   Widget build(BuildContext context) {
-    return Expanded(
-      flex: flex,
-      child: const SizedBox.shrink(),
+    assert(
+      (flex == null) ^ (length == null),
```

Comment:
Asserts like this should be in the constructor.
(if it already is, then the assert message should be something about an internal error indicating the constructor is broken or something)